packages/vuelidate/src/core.js

```diff
@@ -391,11 +418,14 @@ function createMetaFields (results, nestedResults, childResults) {
     // reset the root $dirty state
     results.$reset()
     // reset all the children $dirty states
-    Object.values(allResults.value).forEach((result) => {
+    allResults.value.forEach((result) => {
       result.$reset()
     })
   }
 
+  // Ensure that if all child and nester results are $dirty, this also becomes $dirty
+  if (allResults.value.length && allResults.value.every(nr => nr.$dirty)) $touch()
```

Comment:
Not sure. Need to validate.

Reply:
Gave this a quick test, it gets called when we have dynamic rules and we re-create the validation state, but all properties are already dirty.

I think this is not the proper fix, because if the `$dirty` property on each nested result is set to true, this will never set the parent to dirty on its own. 

We can probably remove it.